pyaptly/publish.py

```diff
@@ -129,11 +132,13 @@ def publish_cmd_update(cfg, publish_name, publish_config, ignore_existing=False)
                     continue
                 prefix_to_search = re.sub("%T$", "", snap["name"])
 
-                current_snapshot = [
-                    snap_name
-                    for snap_name in sorted(current_snapshots, key=lambda x: -len(x))
-                    if snap_name.startswith(prefix_to_search)
-                ][0]
+                current_snapshot = None
+                for snap_name in sorted(current_snapshots, key=lambda x: -len(x)):
+                    if snap_name.startswith(prefix_to_search):
+                        current_snapshot = snap_name
+                        break
+                if current_snapshot is None:
+                    lg.warn("Snapshot %s doesn't exist on to-be archived publish %s." % (snap["name"], publish_fullname))
```

Comment:
You'll need to do something more here - possibly abort, as the line below will likely fail otherwise

Reply:
Thanks for the pointer! It doesn't need to abort, it just doesn't archive the non-existing snapshot. IMO a reasonable behavior.